docs/spec/indexing.rst

```diff
@@ -0,0 +1,144 @@
+.. _indexing:
+
+Indexing
+========
+
+Kartothek uses different types of `inverted file indices`_ to enable efficient partition pruning and improve query performance, see also :doc:`efficient_querying` for more hints on how to optimize performance. This section describes the different types of indices, how to create them and how to interact with them
+
+
+Principle in-memory representation
+----------------------------------
+
+All currently supported kartothek index types are inverted indices and are mapping observed values of a given field to a list of partitions where they were observed.
+
+.. ipython:: python
+
+    index_dct = {1: ["table/12345"], 2: ["table/12345", "table/6789"]}
+
+Where, in this example, the value ``42`` is found in exactly one partition which is labeled ``table/partitionA=42/12345``.
+
+Users typically do not interact with indices directly since querying a dataset will automatically load and interact with the indices. For some applications it is still quite useful to interact with them directly.
+
+All indices implement :class:`~kartothek.core.index.IndexBase` which allows the user to interact with the indices in some useful ways.
+
+.. ipython:: python
+
+    from kartothek.core.index import IndexBase
+
+    index = IndexBase(column="FieldName", index_dct=index_dct)
+
+    index.dtype
```

Comment:
Either the output, a one-sentence explanation, or a link to the docs of the respective function would be helpful here.

Reply:
these are a few examples and the index class is linked above with the full docs. I think this is as verbose as it needs to be